module/VuFind/src/VuFind/Exception/HttpDownloadException.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * HTTP download exception
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Exceptions
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Exception;
+
+use Laminas\Http\Headers;
+
+/**
+ * "Format Unavailable" Exception
+ *
+ * @category VuFind
+ * @package  Exceptions
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class HttpDownloadException extends \Exception implements HttpStatusInterface
```

Comment:
Can we add a unit test for this new class, just to keep coverage up? Should be trivially easy. :-)

Reply:
I'm also willing to help if nobody else has time, though I won't get to it right away.

Reply:
I added a simple unit test for `HttpDownloadException`.

Reply:
@mtrojan-ub, it's good to test the exception handling logic of the caching downloader, but I had actually meant to create a VuFindTest\Exception\HttpDownloadException test which simply creates an exception and confirms that the getters return the expected values, so every new file has a corresponding test. As previously offered, if you don't have time for this, I can probably throw it together fairly quickly. :-)

Reply:
OK, so i added a separate HttpDownloadExceptionTest, hope it's better this time :smile: